scripts/minikube-minishift-setup-env.sh

```diff
@@ -39,6 +39,62 @@ setup_kubeconfig() {
     fi
 }
 
+setup_minikube_developer() {
+    pwd=`pwd`
+    certdir=`mktemp -d`
+    cd $certdir
+    shout "Starting to create minikube developer"
+    openssl genrsa -out developer.key 2048
+    openssl req -new -key developer.key -out developer.csr -subj "/CN=developer/O=minikube"
+    openssl x509 -req -in developer.csr -CA ~/.minikube/ca.crt -CAkey ~/.minikube/ca.key -CAcreateserial -out developer.crt -days 500
+    kubectl config set-credentials developer --client-certificate=developer.crt --client-key=developer.key
+    kubectl config set-context developer-minikube --cluster=minikube --user=developer
+    kubectl create -f - <<EOF
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: odo-user
+rules:
+- apiGroups: [""] # “” indicates the core API group
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["apps"]
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["operators.coreos.com"]
+  resources: ["clusterserviceversions"]
+  verbs: ["*"]
+- apiGroups: ["redis.redis.opstreelabs.in"]
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["networking.k8s.io", "extensions"]
+  resources: ["ingresses"]
+  verbs: ["*"]
+- apiGroups: ["route.openshift.io"]
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["apps.openshift.io"]
+  resources: ["*"]
+  verbs: ["*"]
```

Comment:
Does it make sense to add OpenShift resources to a minikube user? In a real world scenario, I think no minikube user will be concerned about OCP resources, but some of our tests rely on the user having access to these resources, and they might show a different output depending on which resource the user has access to.

Reply:
odo fails if it does not have access to these openshift resources, even in minikube. It could be interesting to open an issue, to make odo ignore resources that it cannot access, instead of failing

Reply:
> Does it make sense to add OpenShift resources to a minikube user? In a real world scenario, I think no minikube user will be concerned about OCP resources, but some of our tests rely on the user having access to these resources, and they might show a different output depending on which resource the user has access to.

I think its fine if we add permission for user to access those resources, as tests are relying on that. However I would like here from Team on this. @kadel @feloy @mohammedzee1000 Could you please share your thoughts. I have just pointed out to few people, however it would be great if any one of our team can share their thoughts.

Reply:
@valaparthvi Could you please confirm if its a problem with odo or just odo test?

Reply:
It is a problem with odo. See #5042 for example.

Reply:
> It could be interesting to open an issue, to make odo ignore resources that it cannot access, instead of failing

https://github.com/openshift/odo/issues/5043

Reply:
I don't understand why we would want access to OpenShift specific apiGroups for a user being created on minikube. But, it's not going to hurt, so I won't request a change.

Reply:
That is my concern as well. But if we don't allow access to these apiGroups, we see a different output from that of a kubeadmin user,